### ISSUE TITLE
local-storage

### ISSUE THREAD

**Issue by maxisam** [tomastrajan/angular-ngrx-material-starter#3]
Thanks for sharing. It is a great repo.

I would like to recommend you using [ngrx-store-localstorage](https://github.com/btroncone/ngrx-store-localstorage) instead.

**Comment by tomastrajan** [comment]
That's a good point, will look into that if i find some time. Feel free to submit a PR if you wanna look into that 😉.

**Comment by axmad22** [comment]
local-storage loads in the initial states saved, but on _@ngrx/effects/init_ for root action event it removes feature states that are yet to be loaded in and only the root state gets hydrated. my question does it support lazy-loaded feature states. thanks for sharing

**Comment by tomastrajan** [comment]
I have looked into this and also checked issues in [ngrx-store-localstorage](https://github.com/btroncone/ngrx-store-localstorage) repo. They have similar problems with hydrating initial state for lazy loaded feature reducers. 

The issue with losing initial state was present also in this repo and was solved by following commit by merging initial state on both `INIT` and `UPDATE` (reducers) actions emitted by ngrx every time a new lazy reducer is added.

Now it seems to work as expected and as a bonus a debug DEV only meta reducerwas added to log all actions and the corresponding states.
